python/lsst/ip/isr/isrTask.py

```diff
@@ -1702,10 +1714,34 @@ def run(self, ccdExposure, *, camera=None, bias=None, linearizer=None,
 
         self.roughZeroPoint(ccdExposure)
 
+        # An extra config validation logic that checks for compatibility
+        # among the config options for amp offset correction.
+        if self.config.doApplyAmpOffset and not self.config.doMeasureAmpOffset:
+            raise RuntimeError("doApplyAmpOffset requires doMeasureAmpOffset to be True.")
+
+        # Measure amp offset corrections within the CCD.
+        if self.config.doMeasureAmpOffset:
+            ampOffsetConfig = AmpOffsetConfig()
+            ampOffsetConfig.doMeasureAmpOffset = True
+            if self.config.doApplyAmpOffset:
+                # Correct for amp offsets within the CCD.
+                ampOffsetConfig.doApplyAmpOffset = True
+                self.log.info("Calculating and applying amp offset corrections.")
+            else:
+                ampOffsetConfig.doApplyAmpOffset = False
+                self.log.info("Calculating amp offset corrections without applying them.")
+            ampOffsetTask = self.ampOffset(config=ampOffsetConfig)
+            ampOffsetTask.run(ccdExposure)
+
         # correct for amp offsets within the CCD
+        # TODO: Remove this whole ``if`` clause on DM-45526.
         if self.config.doAmpOffset:
-            self.log.info("Correcting amp offsets.")
-            self.ampOffset.run(ccdExposure)
+            if self.config.doMeasureBackground or self.config.doApplyAmpOffset:
```

Comment:
This check should be in the `IsrTaskConfig.validate`, as that will cause the task to raise before execution (it "fails fast" instead of doing a bunch of work before failing).